python/ray/experimental/util/types.py

```diff
@@ -9,6 +9,23 @@ class _CollectiveOp(Enum):
 
 @PublicAPI
 class ReduceOp(_CollectiveOp):
```

Comment:
I changed the types to this because python has [strict requirements](https://docs.python.org/3/howto/enum.html#restricted-enum-subclassing) on subclassing enum.

Reply:
Could you elaborate more on this? What's the problem and how to solve it?

Reply:
It seems like enums can only extend on enums that does not define any members. So ReduceOp can extend on _CollectiveOp, but AllReduceOp can't extend on ReduceOp if it defines members.